irc.go

```diff
@@ -176,11 +176,58 @@ var stdMemberships = []membership{
 	{'v', '+'}, // voice
 }
 
-func (m *membership) String() string {
-	if m == nil {
-		return ""
+// memberships always sorted by descending membership rank
+type memberships []membership
+
+func (m memberships) Add(availableMemberships []membership, newMembership membership) memberships {
```

Comment:
Can we make this method have a pointer receiver (`func (m *memberships) Add(…)`)? It's a little too easy to ignore by mistake the return value of this function.